x-either/src/X/Control/Monad/Trans/Either.hs

```diff
@@ -46,3 +51,26 @@ joinErrors g1 g2 = mapEitherT (fmap (join . bimap g2 (first g1))) . runEitherT
 --   and that the "outer" error (like a user error) becomes the Right error
 joinErrorsEither :: (Functor m, Monad m) => EitherT e (EitherT f m) a -> EitherT (Either f e) m a
 joinErrorsEither = joinErrors Right Left
+
+
+-- |
+-- Great for testing. Allows you to keep all properties in EitherT all the way to the top.
+-- Great for testIO:
+-- prop_... = testIO . runOrFail $ do <all EitherT here>
+--
+runOrFail :: (Monad m) => EitherT Text m a -> m a
+runOrFail = (=<<) (either (fail . unpack) return) . runEitherT
```

Comment:
very similar to [`orDie`] (https://github.com/ambiata/x/blob/master/x-optparse/src/X/Options/Applicative.hs#L74-L75)

I dont know if i see the value of it as the testing scenario you describe is very similar to

``` Haskell
foo :: (Monad m) => EitherT Text m Property -> m Property
foo = eitherT (return . failWith) return
```

which i dont really have a problem with typing

maybe you can change this to:

``` Haskell
unifyLeft :: (e -> m a) -> EitherT e m a -> m a
unifyLeft = flip eitherT return
```